sweep/fee_bumper.go

```diff
@@ -1893,3 +1854,57 @@ func (t *TxPublisher) handleReplacementTxError(r *monitorRecord,
 
 	return result
 }
+
+// calculateRetryFeeRate calculates a new fee rate to be used as the starting
+// fee rate for the next sweep attempt if the inputs are to be retried. An error
+// is returned when the fee func is nil and created without error, otherwise an
+// error is returned if the fee func cannot be initialized.
+func (t *TxPublisher) calculateRetryFeeRate(
+	r *monitorRecord) (chainfee.SatPerKWeight, error) {
+
+	// Get the fee function, which will be used to decided the next fee rate
+	// to use if the sweeper decides to retry sweeping this input.
+	feeFunc := r.feeFunction
+
+	// When the record is failed before the initial broadcast is attempted,
+	// it will have a nil fee func. In this case, we'll create the fee func
+	// here.
+	//
+	// NOTE: Since the current record is failed and will be deleted, we
+	// don't need to update the record on this fee function. We only need
+	// the fee rate data so the sweeper can pick up where we left off.
+	if feeFunc == nil {
+		f, err := t.initializeFeeFunction(r.req)
+
+		// TODO(yy): The only error we would receive here is when the
+		// pkScript is not recognized by the weightEstimator. What we
+		// should do instead is to check the pkScript immediately after
+		// receiving a sweep request so we don't need to check it again,
+		// which will also save us from error checking from several
+		// callsites.
+		if err != nil {
+			log.Errorf("Failed to create fee func for record %v: "+
+				"%v", r.requestID, err)
+
+			return 0, err
+		}
+
+		feeFunc = f
+	}
+
+	// Since the sweeping tx has been replaced by another party's tx, we
```

Comment:
Q: Is this always the case that a sweep is replaced by another party, I mean we can also just run out of budget or ?